src/ipc/ipc-common/IPCProxy.h

```diff
@@ -64,15 +64,23 @@ class IPCProxy : public WrapperType, public IPCProxyNewBase
                 if (m_localProviderBinder.provider() == nullptr) {
                     auto serviceName = ipc()->serviceName();
                     auto objPath = ipc()->objectPath();
+                    bool serviceFound = false;
                     for (auto& proxy : m_ipcProxies) {
                         auto proxyAdapterIPCBinder = proxy.ipcBinder;
                         if (proxyAdapterIPCBinder != nullptr) {
+                            proxyAdapterIPCBinder->setEnabled(false); // prevent connecting to server in next two steps
```

Comment:
The logic which handles the enabling/disabling of sub-proxies is already implemented in "refreshProvider()". Please move your code their

Reply:
This would require calling refreshProvider first on unconnected proxies or calling connectToServer in refreshProvider. Current decoupling of concerns is weak but this change would make it even weaker. Moreover hidden connectToServer calls are present in multiple methods, which makes it even less predictable and harder to read in case the code is moved there. 
This class requires refactoring which will be covered by different commits, I disagree to do it in this commit. It just makes no sense.

Reply:
No refactoring is needed : https://github.com/Pelagicore/facelift/pull/274/commits/d33eb8873a820fdf6565253be4d1242daca52efc